src/sonic-py-common/sonic_py_common/device_info.py

```diff
@@ -154,6 +154,52 @@ def get_asic_conf_file_path():
     return None
 
 
+def get_path_to_platform_dir():
+    """
+    Retreives the paths to the device's platform directory
+
+    Returns:
+        A string containing the path to the platform directory of the device
+    """
+    # Get platform
+    platform = get_platform()
+
+    # Determine whether we're running in a container or on the host
+    platform_path_host = os.path.join(HOST_DEVICE_PATH, platform)
+
+    if os.path.isdir(CONTAINER_PLATFORM_PATH):
+        platform_path = CONTAINER_PLATFORM_PATH
+    elif os.path.isdir(platform_path_host):
+        platform_path = platform_path_host
+    else:
+        raise OSError("Failed to locate platform directory")
+
+    return platform_path
+
+def get_path_to_hwsku_dir():
+    """
+    Retreives the path to the device's hardware SKU data directory
+
+    Returns:
+        A string, containing the path to the hardware SKU directory of the device
+    """
+    # Get platform and hwsku
+    (platform, hwsku) = get_platform_and_hwsku()
+
+    # Determine whether we're running in a container or on the host
+    platform_path_host = os.path.join(HOST_DEVICE_PATH, platform)
+
+    if os.path.isdir(CONTAINER_PLATFORM_PATH):
+        platform_path = CONTAINER_PLATFORM_PATH
+    elif os.path.isdir(platform_path_host):
+        platform_path = platform_path_host
+    else:
+        raise OSError("Failed to locate platform directory")
+
+    hwsku_path = os.path.join(platform_path, hwsku)
+
+    return hwsku_path
```

Comment:
This function is mostly duplicated logic from `get_path_to_platform_dir()`. Reuse code as follows:

```python
platform_path = get_path_to_platform_dir()

hwsku = get_hwsku()

hwsku_path = os.path.join(platform_path, hwsku)
```

Also refactor `get_paths_to_platform_and_hwsku_dirs()` similarly to eliminate duplicate code.

Reply:
fixed